### DISCUSSION TITLE
yarn.app.mapreduce.am.command-opts参数占用问题

### DISCUSSION THREAD

**Discussion by yanfangwang** [apache/celeborn#2502] [Q&A]
use -Dyarn.app.mapreduce.am.command-opts=org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn 之后，那么原来的yarn.app.mapreduce.am.command-opts=-Xmx1024m测试了一下就设置不进去了，这个问题是如何解决的

**Comment by FMX** [comment]
Try to use `yarn.app.mapreduce.am.resource.mb`.

**Reply by yanfangwang** [reply]
This is equivalent to setting the total memory. Opts can control how much is used in the heap.

**Comment by FMX** [comment]
There is three approaches：
1. set yarn.app.mapreduce.am.command-opts=-Xmx2048m org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn
2. -Dyarn.app.mapreduce.am.command-opts="-Xmx2048m org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn"
3. set in the yarn-site.xml
